codegen/golang.go

```diff
@@ -87,6 +87,13 @@ func WithUnwrapAllEmbeddedStructs() TableOptions {
 	}
 }
 
+// WithValueTypeOverride sets a function that can override the schema type for specific fields. Return `nil` to fall back to default behavior.
+func WithValueTypeOverride(resolver func(reflect.StructField) *schema.ValueType) TableOptions {
```

Comment:
let's do the `resolver func(any) schema.ValueType`?
So that I'm able to pass `codegen.WithValueTypeOverride(time.Diration{})` for instance